tensorly/decomposition/tests/test_parafac2.py

```diff
@@ -488,36 +488,35 @@ def test_update_imputed():
     # Check that _update_imputed works correctly when imputing values according to the reconstructed tensor,
     # i.e. mode='factors'
 
-    tensor = parafac2_to_tensor(random_parafac2_tensor)
-
-    mask = rng.binomial(1, 0.25, size=tensor.shape)
+    slices = parafac2_to_slices(random_parafac2_tensor)
+    slices_masks = [tl.tensor(rng.binomial(1, 0.25, size=T.shape(slice)),dtype=tl.float64) for slice in slices]
 
     imputed_tensor = _update_imputed(
-        tensor_slices=tensor,
-        mask=mask,
+        tensor_slices=slices,
+        mask=slices_masks,
         decomposition=random_parafac2_tensor,
         method="factors",
     )
 
-    assert_allclose(imputed_tensor, tensor)
+    assert_allclose(imputed_tensor, slices)
 
     # Check that _update_imputed works correctly when imputing values according to nanmean of mode-2 slices
 
-    tensor = np.asarray(tensor)
-
-    tensor[mask == 0] == np.nan
-
-    for slice in tensor:
-        slice[slice == 0] = np.nanmean(slice)
+    slices = np.array(slices)
```

Comment:
Hmmm, maybe we want to use `tl.stack` here instead? Just to be sure that we're using TensorLy tensors?

```suggestion
    slices = tl.stack(slices, axis=0)
```

Reply:
Simply omitting this line seems to do the trick regardless of the backend ¯\_(ツ)_/¯.

Reply:
Fabolous!